src/gretel_trainer/relational/backup.py

```diff
@@ -10,7 +10,7 @@
 @dataclass
 class BackupRelationalDataTable:
     primary_key: list[str]
-    invented_table_metadata: Optional[dict[str, str]] = None
+    invented_table_metadata: Optional[dict] = None
```

Comment:
Would it be worth keeping the first `str` and putting `Any` for value types?
I think we don't use any of these values directly, they are only used to populate `InventedTableMetadata`, right?

Reply:
Yeah good call. I feel like `dict` should _be_ `dict[str, Any]` instead of I guess effectively `dict[Any, Any]`.

You're correct that this dataclass is strictly capturing what goes in and out of the backup file, and the values are just used to populate the `InventedTableMetadata` object. I feel like it's useful to have a pretty strict definition of what to expect to see in the backup JSON files, but it does feel a little obnoxious at times since many structures are nearly or even exactly identical to the internal domain objects used more widely in the code.